packages/ui/src/multiplying-architecture/KaotoBridge.tsx

```diff
@@ -88,10 +88,23 @@ export const KaotoBridge = forwardRef<EditorApi, PropsWithChildren<KaotoBridgePr
    * Subscribe to the `entities:updated` event to update the File content.
    */
   useEffect(() => {
-    return eventNotifier.subscribe('entities:updated', (newContent: string) => {
+    const unsubscribeFromEntities = eventNotifier.subscribe('entities:updated', (newContent: string) => {
       props.onNewEdit(new WorkspaceEdit(newContent));
       sourceCodeRef.current = newContent;
     });
+
+    const unsubscribeFromSourceCode = eventNotifier.subscribe('code:updated', (newContent: string) => {
```

Comment:
Subscribing to `code:updated` event so when a new route is added, the new template update gets reflected on the `KaotoEditor` component